smokey/src/main/java/frc/robot/subsystems/EveryBotPickerSubsystem.java

```diff
@@ -0,0 +1,144 @@
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.SparkMaxRelativeEncoder;
+import com.revrobotics.CANSparkMax.*;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+/**
+ * A class intended to model the picker - checked with Simeon on 02/15/2023 name is now 'picker'
+ */
+public class EveryBotPickerSubsystem extends SubsystemBase {
+
+    /* *********************************************************************
+    CONSTANTS
+    ************************************************************************/
+    // expected to be < 1.0 due to encoder granularity being lower for Rev/Neo
+    private static final double everyBotMotorEncoderTicksPerDegree = Constants.RevNeoEncoderTicksPerRevolution / Constants.DegreesPerRevolution;
+
+    // TODO - use something less than 1.0 for testing
+    private static final double neoMotorSpeedReductionFactor = 1.0;
+
+    /* *********************************************************************
+    MEMBERS
+    ************************************************************************/
+    // two matched motors - one for each climber side
+    private CANSparkMax everyBotMotor = new CANSparkMax(Constants.EveryBotPickerMotorCanId, MotorType.kBrushless);
+    private SparkMaxPIDController everyBotPidController;
+    private RelativeEncoder everyBotEncoder;
+    private double kP, kI, kD, kIz, kFF, kMaxOutput, kMinOutput, maxRPM, maxVel, minVel, maxAcc, allowedErr;
+    private boolean motorInitalizedForSmartMotion = false;
+
+    private boolean isEveryBotMotorInverted = true;
+    private double requestedEveryBotMotorSpeed = 0.0;
+
+    /* *********************************************************************
+    CONSTRUCTORS
+    ************************************************************************/
+
+    /**
+    * constructor for EveryBotPickerSubsystem subsystem
+    */
+    public EveryBotPickerSubsystem() {
+      CommandScheduler.getInstance().registerSubsystem(this);
+    }
+
+    /* *********************************************************************
+    PUBLIC METHODS
+    ************************************************************************/
+    /**
+     * A method to set the every bot motor to a certain speed
+     * @param everyBotArmSpeed the speed to run the everyBot arm motor at
+     */
+    public void setPickerSpeed(double everyBotPickerSpeed) {
+      this.requestedEveryBotMotorSpeed = MotorUtils.truncateValue(everyBotPickerSpeed, -1.0, 1.0);
+    }
+   
+    /**
+     * A method to handle periodic processing
+     */
+    @Override
+    public void periodic() {
+      // confirm that the smart motion is setup - no-op after it is setup first time
+      this.initializeMotorsSmartMotion();
+      this.refreshPickerPosition();
+      this.everyBotMotor.set(this.requestedEveryBotMotorSpeed * neoMotorSpeedReductionFactor);
```

Comment:
should we be setting the speed using the PID controller?
The way we did it in YogiShooter was:
PidController.setReference(Rpm, CANSparkMax.ControlType.kVelocity);

Reply:
ok ... lets try that